dockstore-webservice/src/main/java/io/dockstore/webservice/resources/UserResource.java

```diff
@@ -719,6 +727,176 @@ public Limits setUserLimits(@ApiParam(hidden = true) @Auth User authUser,
+    @ApiOperation(value = "Delete a workflow for a registry and repository path.", authorizations = { @Authorization(value = JWT_SECURITY_DEFINITION_NAME) })
+    public void deleteWorkflow(@ApiParam(hidden = true) @Auth User authUser,
+                                   @ApiParam(value = "Git registry", required = true, allowableValues = "GITHUB_COM, GITLAB_COM, BITBUCKET_ORG") @QueryParam("gitRegistry") TokenType gitRegistry,
+                                   @ApiParam(value = "Git repository path", required = true) @QueryParam("repository") String repository) {
+        User foundUser = userDAO.findById(authUser.getId());
+
+        // Get all of the users source control tokens
+        List<Token> scTokens = workflowResource.checkOnBitbucketToken(foundUser)
+                .stream()
+                .filter(token -> Objects.equals(token.getTokenSource(), gitRegistry))
+                .collect(Collectors.toList());
+
+        // Delete workflow for a given repository
+        if (scTokens.size() > 0) {
+            final Token gitToken = scTokens.get(0);
+            final String tokenSource = gitToken.getTokenSource().toString();
+
+            String gitUrl = "git@" + tokenSource + ":" + repository + ".git";
+            LOG.info("Deleting " + gitUrl);
+
+            // Create a workflow stub object if necessary
+            final Optional<BioWorkflow> existingWorkflow = workflowDAO.findByPath(tokenSource + "/" + repository, false, BioWorkflow.class);
+            if (existingWorkflow.isEmpty()) {
+                String msg = "No workflow with path " + tokenSource + "/" + repository + " exists.";
+                LOG.error(msg);
+                throw new CustomWebApplicationException(msg, HttpStatus.SC_BAD_REQUEST);
+            } else {
+                BioWorkflow workflow = existingWorkflow.get();
+                checkUser(foundUser, workflow);
+                if (Objects.equals(workflow.getMode(), WorkflowMode.STUB)) {
+                    elasticManager.handleIndexUpdate(existingWorkflow.get(), ElasticMode.DELETE);
+                    workflowDAO.delete(workflow);
+                } else {
+                    String msg = "The workflow with path " + tokenSource + "/" + repository + " cannot be deleted.";
+                    LOG.error(msg);
+                    throw new CustomWebApplicationException(msg, HttpStatus.SC_BAD_REQUEST);
+                }
+            }
+        }
+    }
+
+    @GET
+    @Timed
+    @UnitOfWork(readOnly = true)
+    @Path("/registries")
+    @Operation(operationId = "getUserRegistries")
+    @ApiOperation(nickname = "getUserRegistries", value = "Get all of the git registries accessible to the logged in user.", authorizations = { @Authorization(value = JWT_SECURITY_DEFINITION_NAME) }, response = TokenType.class, responseContainer = "List")
+    public List<TokenType> getUserRegistries(@ApiParam(hidden = true) @Auth User authUser) {
+        User foundUser = userDAO.findById(authUser.getId());
```

Comment:
Is this necessary? I've seen cases where I think it is, but since you're only using the ID, I'm not sure it is here. Not a big deal, and I wish I knew Hibernate better to understand myself. Comment applies to subsequent methods

Reply:
I guess it is not due to the authUser already having the same id. @denis-yuen ?

Reply:
Well, the `SimpleAuthenticator` is already getting a user from the database, so I'd be thinking no.

Reply:
I think the issue I ran into in the past is that the User object is cached, so the User object you get passed in is not from the same Hibernate session as the one started by `@UnitOfWork`. And that can lead to issues when updating the database if you change anything on the user object. At least that was my guess.